common/src/stack/report-system/command/report/system/tests/test_sles_patch.py

```diff
@@ -0,0 +1,124 @@
+import pytest
+import os
+import stack.api
+import hashlib
+import json
+
+
+# Returns hash of filename
+def get_hash(filename, sles):
+	# SLES15 uses sha256
+	if '15sp' in sles:
+		hasher = hashlib.sha256()
+	else:
+		hasher = hashlib.md5()
+
+	with open(filename, 'rb') as f:
+		hasher.update(f.read())
+
+	return hasher.hexdigest()
+
+
+# Verify SLES patches are installed
+def test_sles_pallet_patched(host, report_output):
+	total_matched = []
+	base_dir = '/export/stack/pallets/SLES/'
+	if not os.path.isdir(base_dir):
+		pytest.skip('No SLES pallet found - skipping patch check')
+	sles_flavors = os.listdir(base_dir)
+	assert sles_flavors
+
+	# Find out where is stack-sles*images*.rpm file(s)
+	result = host.run('find /export/stack/pallets/stacki/ -name "*stack-sles-*.rpm"')
+	RPM = result.stdout.splitlines()
+	assert RPM
+
+	# If the stack-sles*images*.rpm installed?
+	result = host.run('rpm -qa | grep stack | grep images')
+	assert result
+
+	# Test every sles flavor found
+	for sles_flavor in sles_flavors:
+		# Make sure installed patches match what's under /opt/stack/pallet-patches
+		patch_dir = '/opt/stack/pallet-patches'
+		patch_dir_files = []
+		found_source = False
+		for (dir_path, dir_names, filenames) in os.walk(patch_dir):
+			# Only want particular SLES version
+			if sles_flavor in dir_path:
+				file_to_check = None
+				if '15sp' in sles_flavor:
+					file_to_check = 'CHECKSUMS'
+				else:
+					file_to_check = 'content'
+				patch_dir_files += [os.path.join(dir_path, file) for file in filenames if file == file_to_check]
```

Comment:
sles_flavor is in format of `12sp3` since that is the directory name under `/export/stack/pallets/SLES`. So `Path` won't find the right path to grab pattern from. Here is what we are looking for pattern in:
```
frontend-0-0:/opt/stack/pallet-patches # ls
SLES-11sp3-sles11-sles-x86_64  SLES-12sp3-sles12-sles-x86_64
SLES-12sp2-sles12-sles-x86_64  SLES-15sp1-sles15-sles-x86_64
```
So I couldn't get your suggestion to work. Is there another simpler line I can add?

Reply:
Maybe try this:
```
if '15sp' in sles_flavor:
    pattern = f'SLES-{sles_flavor}-*/**/CHECKSUMS'
else:
    pattern = 'fSLES-{sles_flavor}-*/**/content'

patch_dir_files = list(Path(f'/opt/stack/pallet-patches').glob(pattern))
```
I don't have time to actually test but it should be possible to come up with some sort of glob pattern that would work.

Reply:
Worked!